source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -3591,19 +3598,13 @@ export abstract class Input extends CardElement implements IInput {
         }
     }
 
-    protected registerEventsForReadWriteInputStyle(input: HTMLElement | undefined, inputEventHandler: (input: HTMLElement, eventType: Enums.CardEventType) => void) {
-        if (!input) {
-            return;
+    updateVisualState(hover: boolean): void {
+        if (!hover) {
+            // when user is not hovering on the card, we will show input fields as read only view
+            if (this._renderedInputControlElement && this.inputStyle === Enums.InputStyle.RevealOnHover) {
+                this._renderedInputControlElement.classList.add(this.hostConfig.makeCssClassName("ac-inputStyle-revealOnHover"));
```

Comment:
actually I want to add this class "ac-inputStyle-revealOnHover" at render time. and when hover happens I want to identify the input using this class. For this rule to work .ac-adaptiveCard:hover .ac-input.ac-inputStyle-revealOnHover.

So I dont want to remove the class when hover is true.

It have created two classes then , 
1. updateVisualStateOnRender: It will add common "ac-inputStyle-revealOnHover" to all inputs. Input can override to do some extra work like hiding date picker by input.date
2. On card hover this rule .ac-adaptiveCard:hover .ac-input.ac-inputStyle-revealOnHover already takes care of adding/removing certain style like border. But date/time input needs to do some extra work like shw/hide the icons for date/time picker
3. updateVisualStateOnHover: This class has empty implementation in base Input class. Date and Time input overrides to show and hide icon when hover is there.

Let me know if this sound okay to you.